clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -2952,6 +2952,163 @@ check_exit_found(void)
+            gen_instr(TID_A, /*pc=*/1),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_SYSCALL, 1),
+            gen_marker(TID_A, start_marker, 1),
+            gen_instr(TID_A, /*pc=*/10),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_KERNEL_EVENT, 11),
+            gen_instr(TID_A, /*pc=*/101),
+            gen_marker(TID_A, end_marker, 1),
+            gen_instr(TID_A, /*pc=*/2),
+            gen_exit(TID_A),
+        };
+        if (!run_checker(memrefs, true,
+                         { test_type + " trace has extra kernel_event marker",
+                           /*tid=*/TID_A,
+                           /*ref_ordinal=*/10, /*last_timestamp=*/0,
+                           /*instrs_since_last_timestamp=*/3 },
+                         "Failed to catch extra kernel_event marker in " + test_type +
+                             " trace"))
+            return false;
+    }
+    // Extra signal kernel_xfer in the kernel trace.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_marker(TID_A, TRACE_MARKER_TYPE_FILETYPE, file_type),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_CACHE_LINE_SIZE, 64),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_PAGE_SIZE, 4096),
+            gen_instr(TID_A, /*pc=*/1),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_SYSCALL, 1),
+            gen_marker(TID_A, start_marker, 1),
+            gen_instr(TID_A, /*pc=*/101),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_KERNEL_XFER, 102),
+            gen_instr(TID_A, /*pc=*/102),
+            gen_marker(TID_A, end_marker, 1),
+            gen_exit(TID_A),
+        };
+        if (!run_checker(memrefs, true,
+                         { test_type + " trace has extra kernel_xfer marker",
+                           /*tid=*/TID_A,
+                           /*ref_ordinal=*/8, /*last_timestamp=*/0,
+                           /*instrs_since_last_timestamp=*/2 },
+                         "Failed to catch extra kernel_xfer marker in " + test_type +
+                             " trace"))
+            return false;
+    }
+    // Signal immediately after kernel trace.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_marker(TID_A, TRACE_MARKER_TYPE_FILETYPE, file_type),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_CACHE_LINE_SIZE, 64),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_PAGE_SIZE, 4096),
+            gen_instr(TID_A, /*pc=*/1),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_SYSCALL, 1),
+            gen_marker(TID_A, start_marker, 1),
+            gen_instr(TID_A, /*pc=*/10),
+            gen_marker(TID_A, end_marker, 1),
+            gen_marker(TID_A, TRACE_MARKER_TYPE_SYSCALL, 1),
+            // Consecutive kernel trace, for a stronger test.
+            gen_marker(TID_A, start_marker, 1),
+            gen_instr(TID_A, /*pc=*/10),
+            gen_marker(TID_A, end_marker, 1),
+            // The value of the kernel_event marker is set to pc=2, which is the next
```

Comment:
As above, named constants would help.  It's also not immediately clear that these instructions are all size 1.

Reply:
Added a named constant for KERNEL_TRACE_TYPE. Not so sure named constants for each pc would improve clarity. Many other unit tests use the hardcoded pcs for pc continuity checks, and also omit explicit specification of the size=1 default value for brevity.